static/select_seats.js

```diff
@@ -0,0 +1,99 @@
+        if (booking.hasOwnProperty(seat)) {
+            removeSeat(seat);
+        } else {
+            addSeat(seat);
+        }
+    }
+
+    function addSeat(seat) {
+        var seatId = seat.replace("seat-", "");
+        booking[seat] = {id: seatId, value: null};
+        document.querySelector('#' + seat).classList.add('selected-seat');
+        renderBooking();
+    }
+
+    function removeSeat(seat) {
+        var seatId = seat.replace("seat-", "");
+        delete booking[seat];
+        document.querySelector('#' + seat).classList.remove('selected-seat');
+        renderBooking();
+    }
+
+    function selectSeatType(event) {
+        var target = event.target,
+            seatId = target.dataset.id,
+            seatType = target.value;
+        seatKey = "seat-" + seatId;
+
+        booking[seatKey].value = seatType;
+        renderBooking();
+    }
+
+    function renderBooking() {
+        output = "";
+        Object.keys(booking).forEach(function(seatId) {
+            output += renderSeatForm(booking[seatId]);
+        });
+
+        document.querySelector('#booking').innerHTML = output;
+        document.querySelectorAll('#booking select').forEach(function (select) {
+            select.addEventListener('change', selectSeatType);
+        });
+    }
+
+    function renderSeatForm(seat) {
+        var seatId = seat.id,
+            seatType = seat.value,
+            seatObject = allSeats["seat-" + seatId],
+            displayName = seatObject.name,
+            pricing = pricings[seatObject.group];
+
+        function option(seatType, selectedSeatType) {
+            var selectedString = seatType === selectedSeatType ? " selected" : "";
+            return '<option value="' + seatType + '" '+ selectedString + '>' +
+                seatType[0].toUpperCase() + seatType.slice(1) +
+                ' (' + pricing[seatType] + 'kr)' +
+                '</option>';
+        }
+        return '<div><label>' + displayName + ': ' +
+            '<select name="seat_' + seatId + '" data-id="' + seatId + '">' +
+            '<option value=""></option>' +
+            option('normal', seatType) +
+            option('student', seatType) +
+            '</select></label></div>';
+    }
+
+    Object.keys(allSeats).forEach(function(seat) {
+        var element = document.getElementById(seat),
+            seatObject = allSeats[seat];
+        element.addEventListener('mouseover', function() {
+            var pricing = pricings[seatObject.group],
+                info = '<div>' + seatObject.name + '<br>' +
+                'Student: ' + pricing['student'] + 'kr' + '<br>' +
+                'Normal: ' + pricing['normal'] + 'kr' + '<br>' +
+                '</div>';
+
+            document.querySelector('.seat-info').innerHTML = info;
+        });
+
+        element.addEventListener('mouseout', function() {
+            document.querySelector('.seat-info').innerHTML = "";
+        });
+    });
+
+    Promise.resolve(
```

Comment:
Could you expand that a bit? How do you mean I should use `setImmediate` here instead? The reason I'm doing this is to ensure that the first loop gets run before the one in the `.then` statement. Maybe I'm doing it wrong somehow, but without this (or some similar solution), some of the other parts of this JS won't work properly. Some event handlers get added to already taken seats, etc.